### PR TITLE
Np 154 & NP 141 & 167

### PR DESCRIPTION
**Pull request by Petoxic**
## **Ticket**

**JIRA ID:** [NP-154](https://parking-system.atlassian.net/browse/NP-154)
[NP-141](https://parking-system.atlassian.net/browse/NP-141)
[NP-167](https://parking-system.atlassian.net/browse/NP-167)

## Changes proposed in this pull request
- Add parking privilege
- Fix wording 
- Fix access mode logic between Create/Edit and View

please fill all the required fields from below Checklist otherwise reviewer **will not review** this PR
## Checklist
- [x] Screenshot evidence for both IOS and Android
- [x] Total change is less than 500 lines of code

## **Screenshot Evidence**

**Management role**
https://github.com/ParkingSystemWithLPR/Napark-application/assets/70986870/2be203c2-499a-40bc-a6b0-cb4f1a357e63

**Parking privilege**
https://github.com/ParkingSystemWithLPR/Napark-application/assets/70986870/bbb166f1-6e78-4455-a383-4e9149f3e367

[NP-154]: https://parking-system.atlassian.net/browse/NP-154?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[NP-141]: https://parking-system.atlassian.net/browse/NP-141?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[NP-167]: https://parking-system.atlassian.net/browse/NP-167?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: components/input/ParkingZonePrivilegeInput.tsx [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+                <Controller
+                  name={`privilege.${privilegeIndex}.floor`}
+                  control={control}
+                  render={({ field: { onChange, value } }) => (
+                    <DropdownInput
+                      title="Floor"
+                      selectedValue={value}
+                      placeholder="Select floor"
+                      onSelect={onChange}
+                      items={[{ label: "1", value: "1" }]}
```
Comment (TaeCV) [inline components/input/ParkingZonePrivilegeInput.tsx]: ไม่โดน warning เรื่อง key หรอ

File: components/input/ParkingZonePrivilegeInput.tsx [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+                      title="Unit"
+                      placeholder={"Select fee unit"}
+                      onSelect={(value) => onChange(value)}
+                      items={formatEnumtoDropdownItem(PriceRateUnit)}
+                      containerStyle={{ flex: 1 }}
+                    />
+                  )}
+                />
+              </View>
+              {privilegeIndex === zones.length + zoneIndex - 1 &&
```
Comment (TaeCV) [inline components/input/ParkingZonePrivilegeInput.tsx]: `privilegeIndex === zones.length + zoneIndex - 1` can you make this logic to be more readable? e.g. move it to be a function

File: components/ui/ImageContainer.tsx [hunk 3]
```diff
@@ -7,7 +7,7 @@ import Colors from "@/constants/color";
 export type ImageContainerProps = {
   images: string[];
   editable?: boolean;
-  onDelete?: (imageUrl: string) => void;
+  onDelete?: (image: string) => void;
```
Comment (TaeCV) [inline components/ui/ImageContainer.tsx:10]: why do you need to change the wording here?

File: screens/role/ConfigPrivilege.tsx [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+            key={index}
+            onPress={() =>
+              navigation.navigate("ConfigZone", {
+                form: form,
+                mode: ActionMode.EDIT,
+                index: index,
+              })
+            }
+          />
+        ))}
```
Comment (TaeCV) [inline screens/role/ConfigPrivilege.tsx]: Can you make it more clear when we connect to backend? Please do not think that we use mock here but you have to think how we transfer the data from which place to which place

Comment (TaeCV) [reply screens/role/ConfigPrivilege.tsx]: I saw many places like this. Please fix all of them

File: components/ui/ImageContainer.tsx (hunk 3 above)
Comment (ingthanhunt) [reply components/ui/ImageContainer.tsx:10]: Because, it not only imageUrl but also base64 to delete by this function so i decide to change wording kub

Comment (TaeCV) [reply components/ui/ImageContainer.tsx:10]: Ok got it. You mean that the value passing here might be either url or base64

**Pull request merged**